docs/evm_training/evmTask4.md

```diff
@@ -1,106 +1,179 @@
+git clone https://github.com/NomicFoundation/hardhat-boilerplate.git
+cd hardhat-boilerplate
+npm install
 
-### 1. Configure MetaMask to Use the Rinkeby Testnet
+```
 
-Open MetaMask and select the Ethereum account you want to use for the Force Bridge transfer. This should be the same Ethereum account you are using for Layer 2 on Nervos. Then select the "Rinkeby Test Network" from the Network dropdown.
+This is a brief introduction to the project structure of a Hardhat boilerplate.
 
-<img src={useBaseUrl("img/evmTask4-1.png")}  width="40%"/>
+```bash
 
-You will need to get some Testnet ETH from one of the Rinkeby faucets. You can use the [https://rinkebyfaucet.com/](https://rinkebyfaucet.com/) or [MyCrypto Rinkeby Faucet](https://app.mycrypto.com/faucet) to get a small amount of testnet funds.
+├─ hardhat-boilerplate
+│  ├─ contract <-- save contract `.sol` file
+│  │  └─ Token.sol <-- test sol file
+│  ├─ script
+│  │  └ deploy.js <-- deploy contract to network
+│  ├─ test
+│  │  └ Token.js <-- Automated tests contract 
+│  └─ front-end <-- dapp Web frontend 
+│  │
+│  └─ hardhat.config.js <-- hardhat network settings
+```
 
-### 2. Calculate Your Layer 2 Deposit Receiver Address
+## Start Local Network
+Let's start the local network
+```bash
+npx hardhat node
+```
+You can see the following content, which includes the startup state and port number. By default, it is 8545.
+```bash
+Started HTTP and WebSocket JSON-RPC server at http://127.0.0.1:8545/
 
-Nervos' unique architecture allows a single account to have multiple addresses to be created for different purposes. We will be using one of these addresses to streamline the process of moving tokens to Layer 2.
+Accounts
+========
 
-The process flow we mentioned earlier of Ethereum to Force Bridge to Nervos Layer 1 to Nervos Layer 2 is still going on, but the steps can be consolidated to a single action by the end-user. We will walk through the steps of the developer process, which starts with calculating the Layer 2 deposit receiver address for the user.
+WARNING: These accounts, and their private keys, are publicly known.
+Any funds sent to them on Mainnet or any other live network WILL BE LOST.
 
-Open the example code from [code-examples/bridge/index.js](https://github.com/nervosnetwork/layer2-evm-documentation/blob/master/code-examples/bridge/index.js) in an editor of your choosing.
+Account #0: 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266 (10000 ETH)
+Private Key: 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80
 
-Replace `<YOUR_ETHEREUM_ADDRESS>` with the Ethereum address that you wish to receive your Layer 2 ckETH tokens.
+...other content
+```
+## Test your contract
```

Comment:
Consider to switch the order of `Start Local Network` and `Test your contract`? 
1. Test your contract
2. Start Local Network

Reply:
Yes, it's true that testing contracts doesn't require running a local network environment. The current ordering is based on the idea that when users clone the Hardhat boilerplate, deploying the network might give them a clearer understanding of what they can do with the project after cloning it. However, considering that this tutorial primarily focuses on interacting with contracts, it does make sense to place testing the contract at the beginning.